packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -426,24 +432,93 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         }
     }
 
+    private tooltip(x: number, y: number): void {
+        const xPos = this.positionXMove;
+        const timeForX = this.getTimeForX(xPos);
+        let timeLabel: string | undefined = timeForX.toString();
+        if (this.props.unitController.numberTranslator) {
+            timeLabel = this.props.unitController.numberTranslator(timeForX);
+        }
+        const chartWidth = this.lineChartRef.current.chartInstance.width;
+        const arraySize = this.state.xyData.labels.length;
+        const index = Math.max(Math.round((xPos / chartWidth) * (arraySize - 1)), 0);
+        const points: any = [];
+        let zeros = 0;
+        this.state.xyData.datasets.forEach((d: any) => {
+            // In case there are less data points than pixels in the chart,
+            // calculate nearest value.
+            const adjustedY = d.data[index];
+            const rounded = isNaN(adjustedY) ? 0 : (BIMath.round(Number(adjustedY) * 100)/BIMath.round(100));
+            if (this.state.xyData.datasets.length === 1 || rounded > 0) {
+                const point: any = {
+                    label: d.label,
+                    color: d.borderColor,
+                    background: d.backgroundColor,
+                    value: rounded.toString(),
+                };
+                points.push(point);
+            }
+            else {
+                zeros += 1;
+            }
+        });
+        // Sort in decreasing order
+        points.sort((a: any, b: any) => Number(b.value) - Number(a.value));
+
+        // Adjust tooltip position if mouse is close to the bottom of the window
+        let topPos = undefined;
+        let bottomPos = undefined;
+        if (y > window.innerHeight - 270) {
+            bottomPos = window.innerHeight - y;
+        }
+        else {
+            topPos = window.pageYOffset + y + 15;
+        }
+
+        const tooltipData = {
+            title: timeLabel,
+            dataPoints: points,
+            top: topPos,
+            bottom: bottomPos,
+            left: window.pageXOffset + x - 200,
```

Comment:
Are we guaranteed that the tooltip width will not exceed 200? I'm worried that the series name is unconstrained?

Reply:
Yes, the height of the tooltip is constrained to 250px, so this is safe.

Reply:
Right. The solution is to use "right" instead of "left". Now longer texts will make the tooltip grow to the left, where we have way more space.
Fixed.